tests/SprykerEcoTest/Zed/Adyen/Business/FacadeTest.php

```diff
@@ -162,6 +162,46 @@ public function testHandleNotification(): void
         }
     }
 
+    /**
+     * @return void
+     */
+    public function testHandleNotificationWithAuthorizeAfterCapture(): void
+    {
+        $facade = $this->createFacade();
+        $orderTransfer = $this->setUpCommandTest(
+            static::PROCESS_NAME_ADYEN_CREDIT_CARD,
+            static::OMS_STATUS_NEW
+        );
+
+        $notificationsTransferAuthorize = $this->createNotificationsTransfer($orderTransfer, 'AUTHORISATION');
+        $facade->handleNotification($notificationsTransferAuthorize);
+
+        foreach ($this->getSpySalesOrderItems($orderTransfer) as $item) {
+            /** @var \Orm\Zed\Sales\Persistence\SpySalesOrderItem $item */
+            $paymentAdyenOrderItem = $item->getSpyPaymentAdyenOrderItems()->getLast();
+            $this->assertEquals(static::OMS_STATUS_AUTHORIZED, $paymentAdyenOrderItem->getStatus());
+            $this->assertNotEmpty($paymentAdyenOrderItem->getSpyPaymentAdyen()->getPspReference());
+        }
+
+        $notificationsTransferCapture = $this->createNotificationsTransfer($orderTransfer);
+        $facade->handleNotification($notificationsTransferCapture);
+
+        foreach ($this->getSpySalesOrderItems($orderTransfer) as $item) {
```

Comment:
- Avoid code duplication.
- Extract the cycle to the method assertSalesOrderItemsStatus($orderTransfer, $status);.
- Use instead of 189 and 197 lines.